src/mixins/object_origin.mixin.js

```diff
@@ -1,40 +1,55 @@
 (function() {
 
-  var degreesToRadians = fabric.util.degreesToRadians;
+  var degreesToRadians = fabric.util.degreesToRadians,
+      originXOffset = {
+        left: -0.5,
```

Comment:
because we are always working with half width and half height to translate
the points.

i writing tests for all those functions to be sure they give same result as
old one.
 Il 06/giu/2015 05:07, "Juriy Zaytsev" <notifications@github.com> ha
scritto:

> In src/mixins/object_origin.mixin.js
> <https://github.com/kangax/fabric.js/pull/2257#discussion_r31863927>:
>
> > @@ -1,40 +1,55 @@
> >  (function() {
> >
> > -  var degreesToRadians = fabric.util.degreesToRadians;
> > +  var degreesToRadians = fabric.util.degreesToRadians,
> > +      originXOffset = {
> > +        left: -0.5,
>
> These are the only things that look confusing to me. Why -0.5, 0 and 0.5?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kangax/fabric.js/pull/2257/files#r31863927>.
>